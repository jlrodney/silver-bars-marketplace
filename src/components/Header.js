import React from 'react';

const Header = (props) => (
  <div>
    <h1>{props.title}</h1>
    {props.subtitle && <h2>{props.subtitle}</h2>}
  </div>
);

Header.defaultProps = {
  title: 'SilverBars Marketplace',
  subtitle: 'Live order board'
};

export default Header;
