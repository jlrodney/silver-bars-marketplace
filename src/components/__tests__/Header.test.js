import React from 'react';
import { shallow } from 'enzyme';
import Header from '../Header';

describe('Header should render  correctly', () => {
  it('should have headline', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find('h1').text()).toEqual("SilverBars Marketplace");
  })
  it('should have a subheadling', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find('h2').text()).toEqual("Live order board");
  })
});

describe('should render Header correctly', () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
})
