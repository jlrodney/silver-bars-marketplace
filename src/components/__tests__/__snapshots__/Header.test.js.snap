// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <h1>
    SilverBars Marketplace
  </h1>
  <h2>
    Live order board
  </h2>
</div>
`;
